docs/adr/0009_versioning_strategy.md

```diff
@@ -0,0 +1,305 @@
+
+### CI and automation responsibilities
+
+#### Workflow: `create-release-branch`
+
+**Trigger:** Manual workflow dispatch OR automated trigger when preparing new minor release
+
+**Purpose:** Initialize a new persistent release branch for a component's minor version line
+
+**Required Steps:**
+
+* **Branch validation:** Verify that `releases/<component>/v<major>.<minor>` doesn't already exist
+* **Source determination:** Create branch from `main` for new minor version release
+* **Branch creation:** Create `releases/<component>/v<major>.<minor>` branch using OCM bot
+* **VERSION file initialization:** Ensure component's `VERSION` file reflects the target minor version
+* **Protection rules:** Apply branch protection (require PR reviews, status checks)
+* **Notification:** Alert component maintainers of new release branch creation
+
+**Manual Parameters:**
+
+* `component`: Target component name (e.g., `ocm-cli`, `ocm-controller`)
+* `major.minor`: Target version (e.g., `0.30` for `releases/ocm-cli/v0.30`)
+* `source-ref`: Source branch/tag to create from (default: `main`)
+* `initial-patch`: Starting patch version (default: `0`)
+
+**Permissions Required:** OCM Bot token with branch creation and protection rule management
+
+**Validation Checks:**
+
+* Component exists in repository structure
+* Source reference is valid and accessible
+* Version format follows SemVer conventions
+* No conflicting release branch already exists
+
+**Error Handling:**
+
+* Duplicate branch: Provide guidance on using existing branch or incrementing minor version
+* Invalid component: List available components and naming conventions
+* Permission failures: Alert repository administrators
+
+**Note:** OCM root component does not require release branches - it uses tags directly from `main` branch.
+
+#### Workflow: `release-from-branch`
+
+**Trigger:** `push` to `releases/<component>/**` OR manual workflow dispatch
+
+**Required Steps:**
+
+* Pre-flight checks: validate branch naming convention and component existence
+* Path filtering: only trigger if changes detected in component directory or VERSION file
+* Run component-specific tests and verification
+* Read `cli/VERSION` or `kubernetes/controller/VERSION` (fall back to latest component tag if absent)
+* Compute new patch number using release tooling
+* Create annotated tag `<component>/vX.Y.Z` using OCM bot
+* Build and publish artifacts (binaries, container images, component descriptors)
+* **Root component handling (configurable):**
+  * **Default:** Calculate root component SemVer bump and trigger automatic `ocm` release
+  * **Override:** Skip root release if `--suppress-root-release` flag is set (for coordinated releases)
+* Update `/VERSIONS.yaml` via OCM bot PR with new sub-component version and computed root version (if applicable)
+* Failure handling: if VERSIONS.yaml update conflicts, stop pipeline and alert maintainers
```

Comment:
I hate the entire "push back an updated version file into the repository after a release" flow with a passion. Let's not replicate it.